### ISSUE TITLE
Add rake task to detect trailing whitespace in markdown files

### ISSUE THREAD

**Issue by bastelfreak** [voxpupuli/modulesync_config#151]
IMO it is best practice to avoid trailing whitespace. I'm not sure if any of our tasks already can handle this or if there are existing solutions. Any opinions?

**Comment by bastelfreak** [comment]
maybe we also want to think a step further and offer something to automatically remove the whitespaces?

**Comment by jyaworski** [comment]
@bastelfreak isn't this checked for by puppet-lint and rubocop both?

**Comment by bastelfreak** [comment]
Our existing rake tasks seem to ignore it. I never get complaints when I run them local, just my vim does.

**Comment by dhoppe** [comment]
Our current setup already does that. 

**Puppet lint**
```
dhoppe@macbook ~/Projects/puppet/puppet-kafka (git)-[master] % git diff
diff --git a/manifests/init.pp b/manifests/init.pp
index 33edd6a..e3389d5 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -47,7 +47,7 @@
 #
 #
 class kafka (
-  $version        = $kafka::params::version,
+  $version        = $kafka::params::version,
   $scala_version  = $kafka::params::scala_version,
   $install_dir    = $kafka::params::install_dir,
   $mirror_url     = $kafka::params::mirror_url,
```

```
dhoppe@macbook ~/Projects/puppet/puppet-kafka (git)-[master] % bundle exec rake test
manifests/init.pp:50:trailing_whitespace:ERROR:trailing whitespace found
bundle exec rake test  4.78s user 0.27s system 99% cpu 5.077 total
```

**Rubocop**
```
dhoppe@macbook ~/Projects/puppet/puppet-kafka (git)-[master] % git diff
diff --git a/spec/classes/broker_spec.rb b/spec/classes/broker_spec.rb
index a1c4e7b..ef588ef 100644
--- a/spec/classes/broker_spec.rb
+++ b/spec/classes/broker_spec.rb
@@ -3,7 +3,7 @@ require 'spec_helper'
 describe 'kafka::broker', type: :class do
   let :facts do
     {
-      osfamily: 'Debian',
+      osfamily: 'Debian',
       operatingsystem: 'Ubuntu',
       operatingsystemrelease: '14.04',
       lsbdistcodename: 'trusty',
```

```
dhoppe@macbook ~/Projects/puppet/puppet-kafka (git)-[master] % rubocop .
Inspecting 15 files
.....C.........

Offenses:

spec/classes/broker_spec.rb:6:26: C: Trailing whitespace detected.
      osfamily: 'Debian',
                         ^

15 files inspected, 1 offense detected
```

We could use something like `puppet-lint -f manifests/` and `rubocop -a .` as Git hooks, but I am not sure if this is always a good idea.

**Comment by ghoneycutt** [comment]
@bastelfreak puppet-lint is already doing this for us. Closing as this issue is quite old and seems to be resolved. Please re-open if I'm missing something.

**Comment by bastelfreak** [comment]
Personally, I would also like to see our markdown files without trailing whitespace. This is currently not implemented.

**Comment by dhoppe** [comment]
@bastelfreak I think this issue can be closed since it has been solved via #541.
